From: Max Lv <max.c.lv@gmail.com>
Date: Mon, 28 May 2018 21:53:13 -0700
Subject: Add winsock.h to dist target

---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 4efb890..b926e7d 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -116,5 +116,5 @@ include_HEADERS = shadowsocks.h
 
 noinst_HEADERS = acl.h crypto.h stream.h aead.h json.h netutils.h redir.h server.h tls.h uthash.h \
                  cache.h http.h local.h plugin.h resolv.h tunnel.h utils.h base64.h ppbloom.h \
-                 common.h jconf.h manager.h protocol.h rule.h socks5.h udprelay.h
+                 common.h jconf.h manager.h protocol.h rule.h socks5.h udprelay.h winsock.h
 EXTRA_DIST = ss-nat
