From: Roger Shimizu <rogershimizu@gmail.com>
Date: Mon, 9 Jan 2017 23:44:00 +0900
Subject: fix typo in src/manager.c

---
 src/manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/manager.c b/src/manager.c
index 2e84f6b..6f4c514 100644
--- a/src/manager.c
+++ b/src/manager.c
@@ -548,7 +548,7 @@ check_port(struct manager_ctx *manager, struct server *server)
     ss_free(sock_fds);
 
     if (bind_err == -2) {
-        LOGI("port is avaiable but can not be locked");
+        LOGI("port is available but can not be locked");
     }
 
     return bind_err == -1 ? -1 : 0;
