From: Max Lv <max.c.lv@gmail.com>
Date: Tue, 24 Jan 2017 20:53:13 +0800
Subject: Remove _WIN32

---
 src/encrypt.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/encrypt.c b/src/encrypt.c
index 53ac165..e83eb7c 100644
--- a/src/encrypt.c
+++ b/src/encrypt.c
@@ -40,14 +40,8 @@
 #include <polarssl/ctr_drbg.h>
 #include <polarssl/version.h>
 #define CIPHER_UNSUPPORTED "unsupported"
-
 #include <time.h>
-#ifdef _WIN32
-#include <windows.h>
-#include <wincrypt.h>
-#else
 #include <stdio.h>
-#endif
 
 #elif defined(USE_CRYPTO_MBEDTLS)
 
@@ -56,14 +50,8 @@
 #include <mbedtls/ctr_drbg.h>
 #include <mbedtls/version.h>
 #define CIPHER_UNSUPPORTED "unsupported"
-
 #include <time.h>
-#ifdef _WIN32
-#include <windows.h>
-#include <wincrypt.h>
-#else
 #include <stdio.h>
-#endif
 
 #endif
 
