From: Max Lv <max.c.lv@gmail.com>
Date: Mon, 16 Jan 2017 10:33:25 +0800
Subject: Fix typo of Changes

---
 Changes | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Changes b/Changes
index f7ebcc8..4c76a70 100644
--- a/Changes
+++ b/Changes
@@ -6,7 +6,7 @@ shadowsocks-libev (2.6.2-1) unstable; urgency=medium
 
 shadowsocks-libev (2.6.1-1) unstable; urgency=medium
 
-  * Deprecate HTTP/TLS obfusacting.
+  * Deprecate HTTP/TLS obfuscating.
   * Add SIP003 plugin support.
 
  -- Max Lv <max.c.lv@gmail.com>  Sun, 08 Jan 2017 15:14:19 +0800
