From: Max Lv <max.c.lv@gmail.com>
Date: Mon, 9 Jan 2017 17:15:40 +0800
Subject: Fix #1050

---
 src/server.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/server.c b/src/server.c
index 563d71a..2af10d3 100644
--- a/src/server.c
+++ b/src/server.c
@@ -1877,13 +1877,15 @@ main(int argc, char **argv)
             snprintf(server_str + len, buf_size - len, "|%s", server_host[i]);
             len = strlen(server_str);
         }
+
+        server_host[0] = "127.0.0.1";
+        server_num = 1;
+
         int err = start_plugin(plugin, server_str,
                 plugin_port, server_host[0], server_port);
         if (err) {
             FATAL("failed to start the plugin");
         }
-
-        server_num = 1;
     }
 
     // initialize listen context
