From: Max Lv <max.c.lv@gmail.com>
Date: Thu, 29 Dec 2016 11:40:07 +0800
Subject: Disable fast_open when setsockopt failed

---
 src/server.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/server.c b/src/server.c
index 636d52a..ec6d9c9 100644
--- a/src/server.c
+++ b/src/server.c
@@ -360,6 +360,7 @@ setfastopen(int fd)
         if (s == -1) {
             if (errno == EPROTONOSUPPORT || errno == ENOPROTOOPT) {
                 LOGE("fast open is not supported on this platform");
+                fast_open = 0;
             } else {
                 ERROR("setsockopt");
             }
