From: Simon Shi <simonsmh@gmail.com>
Date: Mon, 16 Jan 2017 18:14:29 -0600
Subject: Update ss-manager.asciidoc

---
 doc/ss-manager.asciidoc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/ss-manager.asciidoc b/doc/ss-manager.asciidoc
index ba5e17f..5c817e6 100644
--- a/doc/ss-manager.asciidoc
+++ b/doc/ss-manager.asciidoc
@@ -15,7 +15,7 @@ SYNOPSIS
  [-b <local_addr>] [-a <user_name>]
  [--manager-address <path_to_unix_domain>]
  [--executable <path_to_server_executable>]
- [--plugin <plugin_name>] [--plugin_opts <plugin_options]
+ [--plugin <plugin_name>] [--plugin_opts <plugin_options>]
 
 DESCRIPTION
 -----------
