From: Max Lv <max.c.lv@gmail.com>
Date: Tue, 10 Jan 2017 11:00:40 +0800
Subject: Fix #1055

---
 src/server.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/server.c b/src/server.c
index e784965..2969f59 100644
--- a/src/server.c
+++ b/src/server.c
@@ -118,6 +118,7 @@ static int nofile = 0;
 static int remote_conn = 0;
 static int server_conn = 0;
 
+static char *plugin          = NULL;
 static char *bind_address    = NULL;
 static char *server_port     = NULL;
 static char *manager_address = NULL;
@@ -1568,7 +1569,6 @@ main(int argc, char **argv)
     char *conf_path = NULL;
     char *iface     = NULL;
 
-    char *plugin      = NULL;
     char *plugin_port = NULL;
     char tmp_port[8];
 
