From: Roger Shimizu <rogershimizu@gmail.com>
Date: Thu, 19 Jan 2017 21:21:36 +0900
Subject: Remove --firewall param in completions/bash/ss-server (#1102)

---
 completions/bash/ss-server | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/completions/bash/ss-server b/completions/bash/ss-server
index 8d089c5..7946a2b 100644
--- a/completions/bash/ss-server
+++ b/completions/bash/ss-server
@@ -1,7 +1,7 @@
 _ss_server()
 {
     local cur prev opts ciphers
-    opts='-s -b -p -k -f -t -m -c -a -n -u -U -v -h -A --fast-open --mtu --help --mptcp -i -6 -d --manager-address --firewall --acl'
+    opts='-s -b -p -k -f -t -m -c -a -n -u -U -v -h -A --fast-open --mtu --help --mptcp -i -6 -d --manager-address --acl'
     ciphers='rc4-md5 table rc4 aes-128-cfb aes-192-cfb aes-256-cfb aes-128-ctr aes-192-ctr aes-256-ctr bf-cfb camellia-128-cfb camellia-192-cfb camellia-256-cfb cast5-cfb des-cfb idea-cfb rc2-cfb seed-cfb salsa20 chacha20 and chacha20-ietf'
     COMPREPLY=()
     cur=${COMP_WORDS[COMP_CWORD]}
