From: Zheng Kai <zhengkai@gmail.com>
Date: Fri, 20 Jan 2017 17:57:58 +0800
Subject: Fix duplicate text

---
 doc/ss-server.asciidoc | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/doc/ss-server.asciidoc b/doc/ss-server.asciidoc
index 19547d9..8c6a246 100644
--- a/doc/ss-server.asciidoc
+++ b/doc/ss-server.asciidoc
@@ -107,11 +107,6 @@ Enable TCP fast open.
 +
 Only available with Linux kernel > 3.7.0.
 
---mptcp::
-Enable Multipath TCP.
-+
-Only available with MPTCP enabled Linux kernel.
-
 --acl <acl_config>::
 Enable ACL (Access Control List) and specify config file.
 
