From: Neal Feng <hellofwy@gmail.com>
Date: Thu, 16 Feb 2017 12:16:44 +0800
Subject: iperf.sh: wait one second before killing processes

---
 scripts/iperf.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/iperf.sh b/scripts/iperf.sh
index 0fd79d1..ac235fc 100755
--- a/scripts/iperf.sh
+++ b/scripts/iperf.sh
@@ -15,6 +15,10 @@ sleep 1
 
 iperf -c 127.0.0.1 -p 8387 -n $number
 
+# Wait for iperf server to receive all data.
+# One second should be enough in most cases.
+sleep 1
+
 kill $ss_tunnel_pid
 kill $ss_server_pid
 kill $iperf_pid
