From: Max Lv <max.c.lv@gmail.com>
Date: Mon, 27 Mar 2017 15:08:55 +0800
Subject: Increase minimum version of cmake

---
 CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 17cfcad..5402b68 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,8 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
-
-# Prevent warning, atleast we need 2.8 anyway.
-cmake_policy(VERSION 2.8)
-cmake_policy(SET CMP0048 OLD)
+cmake_minimum_required(VERSION 3.2)
 
 set(PROJECT_NAME shadowsocks-libev)
 set(RELEASE_DATE 2017-3-21)
