From: Max Lv <max.c.lv@gmail.com>
Date: Wed, 5 Apr 2017 15:12:37 +0800
Subject: Fix #1430

---
 src/utils.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/utils.c b/src/utils.c
index 784c7b5..c3bafcb 100644
--- a/src/utils.c
+++ b/src/utils.c
@@ -37,6 +37,7 @@
 
 #include <sodium.h>
 
+#include "crypto.h"
 #include "utils.h"
 
 #ifdef HAVE_SETRLIMIT
@@ -300,6 +301,10 @@ usage()
         "                                  camellia-256-cfb, bf-cfb,\n");
     printf(
         "                                  chacha20-ietf-poly1305,\n");
+#ifdef FS_HAVE_XCHACHA20IETF
+    printf(
+        "                                  xchacha20-ietf-poly1305,\n");
+#endif
     printf(
         "                                  salsa20, chacha20 and chacha20-ietf.\n");
     printf(
