From: Simon Shi <simonsmh@users.noreply.github.com>
Date: Thu, 6 Apr 2017 21:08:21 -0500
Subject: Update ss-tunnel.asciidoc

---
 doc/ss-tunnel.asciidoc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/doc/ss-tunnel.asciidoc b/doc/ss-tunnel.asciidoc
index 4160208..e6d3667 100644
--- a/doc/ss-tunnel.asciidoc
+++ b/doc/ss-tunnel.asciidoc
@@ -12,9 +12,9 @@ SYNOPSIS
  [-s <server_host>] [-p <server_port>] [-l <local_port>]
  [-k <password>] [-m <encrypt_method>] [-f <pid_file>]
  [-t <timeout>] [-c <config_file>] [-i <interface>]
- [-b <local_addr>] [-a <user_name>] [-n <nofile>]
+ [-b <local_address>] [-a <user_name>] [-n <nofile>]
  [-L addr:port] [--mtu <MTU>] [--mptcp] [--reuse-port]
- [--plugin <plugin_name>] [--plugin-opts <plugin_options]
+ [--plugin <plugin_name>] [--plugin-opts <plugin_options>]
  [--key <key_in_base64>]
 
 DESCRIPTION
