From: Roger Shimizu <rogershimizu@gmail.com>
Date: Sat, 11 Jun 2016 10:56:19 +0900
Subject: use libcorkipset library in system

---
 src/acl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/acl.c b/src/acl.c
index 0851c03..ff759bc 100644
--- a/src/acl.c
+++ b/src/acl.c
@@ -20,7 +20,7 @@
  * <http://www.gnu.org/licenses/>.
  */
 
-#include <ipset/ipset.h>
+#include <libcorkipset/ipset.h>
 #include <ctype.h>
 
 #include "rule.h"
